packages/eslint-plugin/docs/rules/require-await.mdx

```diff
@@ -1,5 +1,5 @@
 ---
-description: 'Disallow async functions which have no `await` expression.'
+description: 'Disallow async functions which do not return promises and have no `await` expression.'
```

Comment:
i guess this could be slightly confusing because every `async` function reuturns a promise. 

I.e. is there a way to be clear that you're talking about the actual `return` statements
```ts
return Promise.resolve();
```
and not the function signature return type
```ts
async function f(): Promise<void>
```
?

Reply:
I can't think of a good wording without turning it into a full documentation, so I think it's fine as-is. The other interpretation is obviously nonsensical so people hopefully won't interpret it that way.

Reply:
Fair enough! 😄